src/main.rs

```diff
@@ -41,16 +48,29 @@ fn handle_search(settings: &Settings, history: &History) {
     let result = Interface::new(settings, history).display();
     if let Some(cmd) = result.selection {
         if let Some(path) = &settings.output_selection {
-            // Output selection to a file, with the first line indicating if the user chose to run the selection or not.
+            // Output selection results to a file.
             let mut out: String = String::new();
 
+            // First we say the desired mode, depending on the key pressed by the user - simply
+            // displaying the selected command, or running it.
             if result.run {
-                out.push_str("run\n");
+                out.push_str("mode run\n");
             } else {
-                out.push_str("display\n");
+                out.push_str("mode display\n");
             }
 
+            // Next, the desired commandline selected by the user.
+            out.push_str("commandline ");
             out.push_str(&cmd);
+            out.push('\n');
+
+            // Finally, any requests for deletion of commands from shell history, for cases where
+            // shells need to handle this natively instead of through us editing HISTFILE.
+            for delete_request in result.delete_requests {
+                out.push_str("delete ");
+                out.push_str(&delete_request);
```

Comment:
What happens if it's a multi-line command?

Reply:
It deletes part of it from from the McFly database, and I think nothing from fish history :/  fish stores multi-line commands on a single line, with "\n" where appropriate, so it's clearer than zsh and I think it can be made to work.  See next comment though...